### PR TITLE
Security updates

### PR DESCRIPTION
**Pull request by nickdala**
PostgreSQL Managed Identity
Front Door rate limiting
Jump box - Microsoft Entra ID for SSH

### CODE REVIEW

File: infra/scripts/setup-service-connector.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+source_id=$1
+target_id=$2
+
+random_string=$RANDOM
+connection_name=postgresql_${RANDOM}
```
Comment (coderandhiker) [inline infra/scripts/setup-service-connector.sh:7]: the connection name has a 50-character maximum; we should ensure this string doesn't exceed it

File: infra/shared/terraform/modules/vms/main.tf [hunk 2]
```diff
@@ -31,8 +35,8 @@ resource "azurerm_linux_virtual_machine" "linux_vm" {
 
   source_image_reference {
     publisher = "Canonical"
-    offer     = "UbuntuServer"
-    sku       = "18.04-LTS"
+    offer     = "0001-com-ubuntu-server-focal"
+    sku       = "20_04-lts-gen2"
```
Comment (coderandhiker) [inline infra/shared/terraform/modules/vms/main.tf:39]: Should we be using 24.04 LTS? that's the latest LTS if we're upgrading anyway

**Review by coderandhiker — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
 			"VARIANT": "jammy"
 		}
 	},
+	"postStartCommand": "./.devcontainer/scripts/onCreateCommand.sh",
 	"remoteUser": "vscode",
 	"containerEnv": {
 		"M2": "/home/vscode" // required because the java feature is not setting this correctly
```
Comment (mnriem) [inline .devcontainer/devcontainer.json:12]: Did you tell the owner of the feature?

File: .devcontainer/scripts/onCreateCommand.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+set -x
+
+az extension add --name serviceconnector-passwordless --upgrade
```
Comment (mnriem) [inline .devcontainer/scripts/onCreateCommand.sh:3]: Why is the --upgrade needed here?

File: infra/scripts/setup-service-connector.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+az webapp connection create postgres-flexible --connection ${connection_name} --source-id ${source_id} --target-id ${target_id} --client-type springBoot --system-identity
```
Comment (mnriem) [inline infra/scripts/setup-service-connector.sh:9]: Any reference here to the Azure CLI command specifically why is springBoot client type needed here?

**Review by mnriem — APPROVED** [review]

File: .devcontainer/scripts/onCreateCommand.sh (hunk 4 above)
Comment (nickdala) [reply .devcontainer/scripts/onCreateCommand.sh:3]: I'll add the following link to this script.

https://learn.microsoft.com/azure/service-connector/tutorial-passwordless?tabs=user%2Cdotnet%2Csql-me-id-dotnet%2Cappservice&pivots=postgresql#install-the-service-connector-passwordless-extension

File: infra/scripts/setup-service-connector.sh (hunk 5 above)
Comment (nickdala) [reply infra/scripts/setup-service-connector.sh:9]: Referenced the following in the script.

https://learn.microsoft.com/azure/service-connector/tutorial-passwordless?tabs=user%2Cdotnet%2Csql-me-id-dotnet%2Cappservice&pivots=postgresql#create-a-passwordless-connection

**Pull request merged**